jscomp/main/bsb_parse_depend.ml

```diff
@@ -48,33 +55,32 @@ let parse_deps_exn lines =
     | Some (_basename, deps) ->
       extract_blank_separated_words deps
 
-let parse_depends ~cwd:_ ~hash files =
- let buf = Buffer.create 1024 in
- Ext_list.iter files (fun file ->
+let single_file file =
   let chan = open_in_bin file in
   let deps = parse_deps_exn (input_lines chan) in
   close_in chan;
-  Buffer.add_string buf "\n(rule (targets ";
-  Buffer.add_string buf (Ext_filename.new_extension file Literals.suffix_depends);
-  Buffer.add_string buf ")\n (action (write-file %{targets} ";
-  Buffer.add_string buf hash;
-  Buffer.add_string buf "))\n ";
-  if deps <> [] then begin
-  Buffer.add_string buf "(deps";
-    Ext_list.iter deps (fun dep ->
-      Buffer.add_string buf Ext_string.single_space;
-      Buffer.add_string buf dep);
-    Buffer.add_string buf ")"
-  end;
-  Buffer.add_string buf ")\n");
- Bsb_ninja_targets.revise_dune Literals.dune_bsb_inc buf
+  let rules = List.map (fun file -> D.read_file ~path:(P.of_string file))
+  deps
+  in
+  List.fold_left (fun acc item ->
+    let+ _ = D.both acc item in ())
+    (D.return ())
+    rules
```

Comment:
```suggestion
  List.fold_left D.both rules |> D.map ~f:(fun _ -> ())
```

Is this the same?

Reply:
I think I tried this and the compiler wasn't happy with "the type variable 'a occurs inside ..."

Reply:
```
Error: This expression has type 'a D.t -> 'b D.t -> ('a * 'b) D.t
       but an expression was expected of type 'a D.t -> 'b D.t -> 'a D.t
       The type variable 'a occurs inside 'a * 'b
```

Reply:
Yeah that's not going to work, will open a following PR to refactor it